### PR TITLE
[Woo POS] Animated transition to the Get Support screen

### LINKED ISSUE #12628: [Woo POS] Animate transition to "Get support" screen
The goal is to update the enter and exit animation to match design specs.

Animation specs: https://wp.me/p6riRB-aNq

### PR DESCRIPTION
**Pull request by samiuelson**
Closes: #12628

### Description

This PR attempts to animate the transition to the Get Support screen as per design feedback and animations specs (p6riRB-aNq-p2).

Screenshot 2024-09-17 at 12 44 58

### Steps to reproduce

1. Enter POS mode
2. Tap "Get Support" from the bottom menu
3. Observe transition to the support screen

### Testing information

Transitions should follow animation specs (see above), and animate by sliding horizontally between screens.

### The tests that have been performed

As shown in the section below, I checked the animations of transitions when entering and exiting the "Get Support" screen. In the store management mode, the transitions should remain unchanged.

Devices: Galaxy Tab S9 FE, Android 14

#### Before
https://github.com/user-attachments/assets/593f4cd2-c295-44f0-b3e7-d764713f3e04

#### After
https://github.com/user-attachments/assets/bca0190d-5998-415b-b700-aa4494eb2c21

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt [hunk 1]
```diff
@@ -55,6 +57,40 @@ class SupportRequestFormActivity : AppCompatActivity() {
...
+                )
+            } else {
+                overridePendingTransition(R.anim.woopos_slide_in_left, R.anim.woopos_slide_out_right)
+            }
+        }
+    }
+
+    @Suppress("DEPRECATION")
+    private fun isPOS(): Boolean {
+        val origin = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt]: @samiuelson 👋

Take a look into `DataContainerExt.kt` file - I think we have extensions there

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt [hunk 2]
```diff
@@ -55,6 +57,40 @@ class SupportRequestFormActivity : AppCompatActivity() {
...
         }
         viewModel.onViewCreated()
+
+        adjustActivityTransitions()
+    }
+
+    @Suppress("DEPRECATION")
+    private fun SupportRequestFormActivity.adjustActivityTransitions() {
+        if (isPOS()) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt]: The same `SystemVersionUtils`. Not sure though if we want to reuse both in the POS

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt [hunk 3]
```diff
@@ -55,6 +57,40 @@ class SupportRequestFormActivity : AppCompatActivity() {
...
+                overrideActivityTransition(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt:72]: Is this purposefully that we we kinda going back (from left to right?). Shouldn't it be from right to left to show that we go forward here? If that's true, then there

```
android.R.anim.slide_in_left
android.R.anim.slide_out_right
```

to be reused

**Comment by backwardstruck** [comment]
I think this is the issue you meant to link:
https://github.com/woocommerce/woocommerce-android/issues/12628

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt (hunk 3 above)
Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt:72]: That seems to be a reasonable conclusion. I don't think it's actually specified in the design (unless I didn't see it).

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/237">20.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt (hunk 1 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt]: Updated — [d7548d1](https://github.com/woocommerce/woocommerce-android/pull/12630/commits/d7548d112609642d31726fbaa586c85e7ef1f2c6)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt (hunk 2 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt]: Done — [7a8b219](https://github.com/woocommerce/woocommerce-android/pull/12630/commits/7a8b219f0ddde321e61043f37f44f36bf69a0a74)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt (hunk 3 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/support/requests/SupportRequestFormActivity.kt:72]: Done — [afc7e26](https://github.com/woocommerce/woocommerce-android/pull/12630/commits/afc7e26cfcdd8a5f842ca338985d26c0b10bd54c) and [4a790bb](https://github.com/woocommerce/woocommerce-android/pull/12630/commits/4a790bbd76a28514a86bdc7e60abd1fbf0a48896)

**Comment by samiuelson** [comment]
Thanks for review, @kidinov! I updated the animations:

https://github.com/user-attachments/assets/4ef35431-86ec-4638-967a-495ead9b3421

Let me know what you think.

**Comment by kidinov** [comment]
@samiuelson, my understanding is that we should have something like this (no alpha, just moving from the right to left and backwards). The animations from android do have alpha so custom are needed

Here is the patch for android 34+
[Animation_for_34+.patch](https://github.com/user-attachments/files/17046084/Animation_for_34%2B.patch)

https://github.com/user-attachments/assets/36c633c1-99a0-42a0-b3ae-9beb404dc255

For Android <34, we also need to add `overridePendingTransition` in the activity's finish method to have proper back animation

**Comment by samiuelson** [comment]
> @samiuelson, my understanding is that we should have something like this (no alpha, just moving from the right to left and backwards). The animations from android do have alpha so custom are needed

Makes sense, updated — [278c0b8](https://github.com/woocommerce/woocommerce-android/pull/12630/commits/278c0b8d0b2828b08494b07da80523cf798aead0)

> For Android <34, we also need to add `overridePendingTransition` in the activity's finish method to have proper back animation

Good catch, fixed:

https://github.com/user-attachments/assets/7541aee2-8468-40c8-9aa7-08e7528198ff

Let me know what you think, @kidinov.

**Comment by samiuelson** [comment]
👋🏼 @backwardstruck, @kidinov, this is ready for review again.

**Review by kidinov — COMMENTED** [review]
> * Can we just override `finish` method instead of calling `adjustExitTransition` from multiples places? I haven't tried it, but I recall that was the way
> 
> ```
>     override fun finish() {
>         adjustExitTransition()
>         super.finish()
>     }
> ```

Good catch, thanks @kidinov ! I think it should be called right after `super.finish()`; Fixed – [a1fe6af](https://github.com/woocommerce/woocommerce-android/pull/12630/commits/a1fe6afa2df4007ae86270112218595c43b1afad)

> I think on Android <34 it slides in the opposite direction it should (from left to right)

Fixed – [efb7e35](https://github.com/woocommerce/woocommerce-android/pull/12630/commits/efb7e3519701be1c1956099db9858a1ef2c80ef1)

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/238">20.6</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by kidinov — APPROVED** [review]
Looks great!

I removed unused import to make it mergable 

a713b1ae17064ba169c47a28bf5c45b368232914

**Pull request merged**